test-cypress/integration/sparql/sparql.menu.spec.js

```diff
@@ -1,807 +1,1113 @@
+
+    function getActiveTabLink() {
+        return cy.get(TABS_SELECTOR + '.active .nav-link')
+    }
+
+    function getNewTabButton() {
+        return cy.get('#wb-sparql-addNewTab');
+    }
+
+    function addTab() {
+        getNewTabButton().click();
+    }
+
+    function getLastTab() {
+        return getTabs().last();
+    }
+
+    function renameTab(position, newName) {
+        getTabs().eq(position - 1).then(tab => {
+            cy.wrap(tab)
+            // First click is to focus it in case it's not the active tab
+                .click()
+                .find('.nav-link')
+                .dblclick();
+            cy.wrap(tab)
+                .find('.editable-input')
+                .type(newName)
+                .should('have.value', newName);
+            cy.wrap(tab)
+                .find('.editable-buttons .btn-primary')
+                .click()
+                .should('not.exist');
+        });
+    }
+
+    function verifyTabName(position, name) {
+        getTabs().eq(position - 1)
+            .should('be.visible')
+            .find('.nav-link')
+            .should('have.text', name);
+    }
+
+    function getTabCloseBtn(position) {
+        return getTabs().eq(position - 1).find('.delete-sparql-tab-btn');
+    }
+
+    function closeTab(position) {
+        getTabCloseBtn(position).click();
+    }
+
+    function verifyQueryAreaContains(query) {
+        // Using the CodeMirror instance because getting the value from the DOM is very cumbersome
+        getQueryArea().should(codeMirrorEl => {
+            const cm = codeMirrorEl[0].CodeMirror;
+            expect(cm.getValue().includes(query)).to.be.true;
+        });
+    }
 
+    function verifyQueryAreaEquals(query) {
+        // Using the CodeMirror instance because getting the value from the DOM is very cumbersome
+        getQueryArea().should(codeMirrorEl => {
+            const cm = codeMirrorEl[0].CodeMirror;
+            expect(cm.getValue().trim()).to.equal(query.trim());
+        });
+    }
 
-        // Verify that there are two results displaying the links between ":a owl:sameAs :a" and ":a owl:sameAs :b"
-        cy.verifyQueryResultsS(2);
+    function waitUntilQueryIsVisible() {
+        getQueryArea().should(codeMirrorEl => {
+            const cm = codeMirrorEl[0].CodeMirror;
+            expect(cm.getValue().trim().length > 0).to.be.true;
+        });
+    }
+
+    function getQueryArea() {
+        return cy.get('#queryEditor .CodeMirror');
+    }
+
+    function getQueryTextArea() {
+        return getQueryArea().find('textarea');
+    }
+
+    function getNoQueryRun() {
+        return cy.get('#yasr-inner .no-query-run');
+    }
+
+    function clearQuery() {
+        // Using force because the textarea is not visible
+        getQueryTextArea().type('{ctrl}a{backspace}', {force: true});
+    }
+
+    function typeQuery(query) {
+        clearQuery();
+        getQueryTextArea().type(query, {force: true, parseSpecialCharSequences: false});
```

Comment:
I guess all `force:true ` are needed everywhere where the hidden textarea is operated?

Reply:
Yes